crates/derive/src/pipeline/core.rs

```diff
@@ -95,6 +106,8 @@ where
         match signal {
             s @ Signal::Reset(ResetSignal { l2_safe_head, .. }) |
             s @ Signal::Activation(ActivationSignal { l2_safe_head, .. }) => {
+                self.metrics.inc_reset_signals();
```

Comment:
What do you think about using the `record_` syntax for most metrics?

e.g. instead of `inc_reset_signals()` and `inc_flush_channel_signals()`, we could use a more generic `record_signal(Signal)` method like you did below with `record_step_result(StepResult)`

Reply:
This allows the metrics implementation to have more flexibility with the metric types (whether it is an increment or an observation etc etc). It may also allow for better forwards compatibility; imagine if there is a new `Signal` variant, the metrics implementation may use a histogram with the signal variants as labels, requiring no changes to be made to neither the metrics interface nor the metrics impl.